File: 백준/Bronze/2884. 알람 시계/알람 시계.py
import sys
input = sys.stdin.readline

n = list(map(int, input().split()))

if n[1] >= 45:
  print(n[0], n[1]-45)
elif n[0]==0:
  print(23, 60+n[1]-45)
else: print(n[0]-1, 60+n[1]-45)

File: 백준/Bronze/2884. 알람 시계/알람 시계.js
const fs = require("fs");
const filePath = process.platform === "linux" ? 0 : "run/input.txt";
const input = fs.readFileSync(filePath, "utf-8").toString().split(" ");

const calTime = (num) => {
  const arr = num.map((elem) => {
    return Number(elem);
  });

  if (arr[0] === 0) {
    if (arr[1] < 45) {
      console.log(23, 15 + arr[1]);
    } else {
      console.log(0, arr[1] - 45);
    }
  } else if (arr[1] < 45) {
    console.log(arr[0] - 1, 15 + arr[1]);
  } else if (arr[1] >= 45) {
    console.log(arr[0], arr[1] - 45);
  }
};

calTime(input);
